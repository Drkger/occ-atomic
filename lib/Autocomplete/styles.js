import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
