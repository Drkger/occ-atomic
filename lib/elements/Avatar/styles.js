export default {
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
};