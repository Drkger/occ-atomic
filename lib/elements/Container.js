import React from 'react';
import injectSheet from 'react-jss';

const styles = {
    containerFluid: {
        paddingRight:'15px',
        paddingLeft:'15px',
        marginRight:'auto',
        marginLeft:'auto'
    },
    container: {
        extend: 'containerFluid',
    },
    '@media (min-width: 768px)': {
        container: {
            width:'750px'
        }
    },
    '@media (min-width: 992px)': {
        container: {
            width:'970px'
        }
    },
}

const Container = ({ classes, children, fluid, extraClasses }) => (
    <div className={`${fluid ? classes.containerFluid : classes.container} ${extraClasses ? extraClasses : ''}`}>{children}</div>
);

export default injectSheet(styles)(Container);