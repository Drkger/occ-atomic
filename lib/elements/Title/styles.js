import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';

export default {
    title: {
        boxSizing:'border-box',
        fontFamily:fonts.title,
        color:colors.title,
        fontWeight:'600',
        marginTop:'20px',
        marginBottom:'15px',
        lineHeight:'1.1'
    },
    h1: {
        fontSize:'22px'
    },
    h2: {
        fontSize:'20px',
    },
    h3: {
        fontSize:'18px',
    },
    h4: {
        fontSize:'15px'
    }
};