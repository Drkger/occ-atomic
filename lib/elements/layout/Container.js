import React from 'react';
import injectSheet from 'react-jss';

import grid from '../../subatomic/grid';

const styles = {
    conFluid: {
        boxSizing:'border-box',
        paddingRight:'15px',
        paddingLeft:'15px',
        marginRight:'auto',
        marginLeft:'auto',
        '&:before, &:after': {
            display:'table',
            content:'""'
        },
        '&:after': {
            clear:'both'
        }
    },
    con: {
        extend: 'conFluid',
        margin:'0 auto'
    },
    [`@media (min-width: ${grid.sm}px)`]: {
        con: {
            width:'750px'
        }
    },
    [`@media (min-width: ${grid.md}px)`]: {
        con: {
            width:'970px'
        }
    },
    [`@media (min-width: ${grid.lg}px)`]: {
        con: {
            width:'1170px'
        }
    }
}

const Container = ({ classes, children, fluid, className, style }) => (
    <div className={`${fluid ? classes.conFluid : classes.con} ${className ? className : ''}`} style={style}>{children}</div>
);

export default injectSheet(styles)(Container);