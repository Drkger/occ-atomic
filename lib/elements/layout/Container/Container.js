import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

const Container = ({ classes, children, fluid, className, id, style }) => (
    <div className={`${fluid ? classes.conFluid : classes.con}${className ? ` ${className}` : ''}`} style={style} id={id}>{children}</div>
);

Container.propTypes = {
    classes: PropTypes.object,
    children: PropTypes.node,
    fluid: PropTypes.bool,
    className: PropTypes.string,
    id: PropTypes.string,
    style: PropTypes.object
};

export default Container;