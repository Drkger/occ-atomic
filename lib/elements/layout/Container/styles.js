import grid from '../../../subatomic/grid';

export default {
    conFluid: {
        boxSizing:'border-box',
        paddingRight:'15px',
        paddingLeft:'15px',
        marginRight:'auto',
        marginLeft:'auto',
        '&:before, &:after': {
            display:'table',
            content:'""'
        },
        '&:after': {
            clear:'both'
        }
    },
    con: {
        extend: 'conFluid',
        margin:'0 auto'
    },
    [`@media (min-width: ${grid.sm}px)`]: {
        con: {
            width:'750px'
        }
    },
    [`@media (min-width: ${grid.md}px)`]: {
        con: {
            width:'970px'
        }
    },
    [`@media (min-width: ${grid.lg}px)`]: {
        con: {
            width:'1170px'
        }
    }
};