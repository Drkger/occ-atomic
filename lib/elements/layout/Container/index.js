import injectSheet from 'react-jss';
import Container from './Container';
import styles from './styles';

export default injectSheet(styles)(Container);