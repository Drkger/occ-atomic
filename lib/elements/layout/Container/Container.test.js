import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';
import styles from './styles';
import ContainerJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Container", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Container classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders a regular container', () => {
        const wrapper = shallow(
            <Container classes={classes} />
        );
        expect(wrapper.find('div').prop('className')).toEqual('con');
    });

    it('renders a fluid container', () => {
        const wrapper = shallow(
            <Container classes={classes} fluid />
        );
        expect(wrapper.find('div').prop('className')).toEqual('conFluid');
    });

});

describe("ContainerJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <ContainerJSS />
        );
        expect(wrapper).toMatchSnapshot();
    });

});