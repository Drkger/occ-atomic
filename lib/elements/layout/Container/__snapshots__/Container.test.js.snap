// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Container
    classes={
      Object {
        "@media (min-width: 1200px)": "@media (min-width: 1200px)",
        "@media (min-width: 768px)": "@media (min-width: 768px)",
        "@media (min-width: 992px)": "@media (min-width: 992px)",
        "con": "con",
        "conFluid": "conFluid",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "con",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "con",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ContainerJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Container) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "con": "Container-con-0-1-2",
        "conFluid": "Container-conFluid-0-1-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "con": "Container-con-0-1-2",
          "conFluid": "Container-conFluid-0-1-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
